src/nu/xom/tests/BuilderTest.java

```diff
@@ -1816,19 +1841,17 @@ public void testInternalAndExternalDTDSubsetWithCrimson()
     // This test requires network access and may be skipped if unavailable
     public void testBaseRelativeResolutionRemotelyWithDirectory()
       throws IOException, ParsingException {
+        if (CITestUtil.isRunningInCI()) {
+            // Skip network tests in CI where connectivity may be unreliable
+            return;
+        }
         try {
             builder.build("http://www.ibiblio.org/xml");
         } catch (IOException e) {
-            // Skip test if network is unavailable - check exception chain
-            Throwable cause = e;
-            while (cause != null) {
-                if (cause instanceof java.net.UnknownHostException ||
-                    cause instanceof java.net.ConnectException) {
-                    // TODO: Use @Ignore or assumeTrue when we move to JUnit 4
-                    systemErr.println("Skipping testBaseRelativeResolutionRemotelyWithDirectory: network unavailable");
-                    return;
-                }
-                cause = cause.getCause();
+            if (CITestUtil.isNetworkException(e)) {
```

Comment:
we don't need this and the if (CITestUtil.isRunningInCI()) check above. I thijk keep this one and remove that one.